publish.ps1

```diff
@@ -0,0 +1,44 @@
+[cmdletbinding()]
+Param(
+  [switch]$Publish
+)
+
+Begin {
+  $BuildFolder = "$PSScriptRoot\Puppet.Dsc"
+  $MarkdownDocsFolder = Resolve-Path "$PSScriptRoot\docs"
+
+  If (Test-Path -Path $BuildFolder) {
+    Remove-Item -Path $BuildFolder -Recurse -Force
+  }
+  New-Item -Path $BuildFolder -ItemType Directory | Out-Null
+}
+
+Process {
+  # Copy source files
+  $FoldersToCopy = @(
+    "$PSScriptRoot\src\en-us"
+    "$PSScriptRoot\src\functions"
+    "$PSScriptRoot\src\internal"
+    "$PSScriptRoot\src\xml"
+  )
+  $FilesToCopy = @(
+    "$PSScriptRoot\src\Puppet.Dsc.psd1"
+    "$PSScriptRoot\src\Puppet.Dsc.psm1"
+    "$PSScriptRoot\src\readme.md"
+  )
+  Copy-Item -Path $FoldersToCopy -Destination $BuildFolder -Recurse
```

Comment:
Comment in general, is what happens when one of these fails. It won't be terminating, and you'll be left with inconsistent state while the rest of the commands continue on blindly. At this stage though, commenting here and creating an issue to address later is sufficient